modules/ting_covers/ting_covers.module

```diff
@@ -144,3 +144,87 @@ function _ting_covers_delete_old_files($time = NULL) {
 function ting_covers_object_path($object_id) {
   return file_default_scheme() . '://ting' . DIRECTORY_SEPARATOR . 'covers' . DIRECTORY_SEPARATOR . 'object' . DIRECTORY_SEPARATOR . md5($object_id) . '.jpg';
 }
+
+/**
+ * Return cover information relatively information given in parameter.
+ *
+ * @param array $cover_data
+ *   Info about cover.
+ *
+ * @return array
+ *   With covers local id, url and style.
+ */
+function ting_covers_prepare_covers(array $cover_data) {
+  $covers = array();
+  $image_styles = array();
+  $missing_images_local_ids = array();
+
+  foreach ($cover_data as $cover_info) {
+    // Determine if the local id is a known negative.
+    if (cache_get('ting_covers:' . $cover_info['local_id'], FALSE)) {
+      break;
+    }
```

Comment:
I have mentioned this before, but shouldn't this be a continue instead of break?

If one of the covers is found to be a known negative, we should ignore that and continue iterating the covers. Not break the loop entirely.